include/mbgl/map/map.hpp

```diff
@@ -58,15 +55,15 @@ class Map : private util::noncopyable {
     // Main render function.
     void render(View&);
 
-    // Styling
-    style::TransitionOptions getTransitionOptions() const;
-    void setTransitionOptions(const style::TransitionOptions&);
-
+    // Style
     void setStyleURL(const std::string&);
```

Comment:
With this method, the workflow is to create the map, then set its style to a URL. #6386 also proposes exposing an  `-[MGLStyle initWithURL:]`. I think this would require a way to initialize an `mbgl::style::Style` with nothing but a URL, so that later on `mbgl::Map::setStyle()` could be called, passing in that style. Do you see any complications with that approach? `-[MGLStyle initWithJSONData:]` (analogous to `setStyleJSON()`) is also a highly requested feature.